app/CreateStore.tsx

```diff
@@ -95,17 +98,21 @@ export function CreateStore(props: {
             if (suburb === "") {
               return setErrorMessage("Suburb cannot be empty");
             }
+            setAwaitingCreateStoreCheck(true);
             if (await displayNameAndSuburbAlreadyExist(displayName, suburb)) {
+              setAwaitingCreateStoreCheck(false);
               return setErrorMessage(
                 "Display name and suburb combination already exists"
               );
             }
             if (await fullNameAndSuburbAlreadyExist(fullName, suburb)) {
+              setAwaitingCreateStoreCheck(false);
               return setErrorMessage(
                 "Full name and suburb combination already exists"
               );
             }
             setErrorMessage("");
+            setAwaitingCreateStoreCheck(false);
             onCreateStoreClick();
             createNewStore(displayName, fullName, suburb);
```

Comment:
if we set this as false here, won't the button be enabled again before the store is created since we haven't even called `createNewStore`? I guess the same also applies to the account creation

Reply:
https://github.com/auberginearms/groceryDatabaseApp/pull/24/commits/f146e2f2836ba87f5eb17170db3e7b75c5bd9892

Reply:
this is better but I think still doesn't work because the backend calls aren't `awaited`. When you create a store and go back to the list, do you see the store you just created in the list?

Reply:
when you create a store, it automatically takes you back to the list and the newly created store is there

Reply:
hmm that's pretty magical oh well we can worry about it when it becomes a bug